### PR TITLE
8234532: Remove ThreadLocalAllocBuffer::_fast_refill_waste since it is never set

### PR DESCRIPTION
**Pull request by albertnetymk**
It has some cascading effect; two performance variables, `fastWaste` and `maxFastWaste`, can be removed also.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8234532](https://bugs.openjdk.java.net/browse/JDK-8234532): Remove ThreadLocalAllocBuffer::_fast_refill_waste since it is never set

### Reviewers
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**) ⚠️ Review applies to 867bbe145bf4aa591dc09c23f53d2faa7dd24984
 * [Stefan Johansson](https://openjdk.java.net/census#sjohanss) (@kstefanj - **Reviewer**)
 * [Per Liden](https://openjdk.java.net/census#pliden) (@pliden - **Reviewer**)
 * [Serguei Spitsyn](https://openjdk.java.net/census#sspitsyn) (@sspitsyn - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3756/head:pull/3756` \
`$ git checkout pull/3756`

Update a local copy of the PR: \
`$ git checkout pull/3756` \
`$ git pull https://git.openjdk.java.net/jdk pull/3756/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3756`

View PR using the GUI difftool: \
`$ git pr show -t 3756`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3756.diff">https://git.openjdk.java.net/jdk/pull/3756.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/shared/threadLocalAllocBuffer.cpp [hunk 1]
```diff
@@ -105,10 +104,9 @@ void ThreadLocalAllocBuffer::accumulate_and_reset_statistics(ThreadLocalAllocSta
     stats->update_fast_allocations(_number_of_refills,
                                    _allocated_size,
                                    _gc_waste,
-                                   _fast_refill_waste,
                                    _slow_refill_waste);
   } else {
-    assert(_number_of_refills == 0 && _fast_refill_waste == 0 &&
+    assert(_number_of_refills == 0 &&
```
Comment (tschatzl) [inline src/hotspot/share/gc/shared/threadLocalAllocBuffer.cpp:109]: Please fill back the clause from the next line. I think the assert condition now fits a single line, the text may be on the next line

**Review by tschatzl — APPROVED** [review]
Lgtm apart from that minor formatting nit.

**Review by kstefanj — APPROVED** [review]
Looks good.

**Review by pliden — APPROVED** [review]
Looks good.

But we should also probably rename `_slow_refill_waste` to `_refill_waste`, since "slow" doesn't really make sense anymore. Of course, that would also affect the `slowWaste` PerfCounter. Maybe this could be a follow up refinement?

**Comment by albertnetymk** [comment]
> Maybe this could be a follow up refinement?

Sure, will do so after this PR.

**Review by sspitsyn — APPROVED** [review]
Hi Albert,
It looks good.
Thanks,
Serguei

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**